### PR TITLE
strided Pool funcs

### PR DESCRIPTION
**Pull request by ryanneph**

### CODE REVIEW

**Comment by ryanneph** [comment]
Implementing on CPU now. Hold on.

File: test/test_ops.py [hunk 1]
```diff
@@ -87,18 +87,22 @@ def test_strided_conv2d(self):
...
-        lambda x: torch.nn.functional.max_pool2d(x, kernel_size=sz),
-        lambda x: Tensor.max_pool2d(x, kernel_size=sz), gpu=self.gpu, forward_only=self.gpu)
-
-  def test_avgpool2x2(self):
-    # TODO Grad tolerance needs to be slightly relaxed; why?
-    helper_test_op([(32,2,111,28)], lambda x: torch.nn.functional.avg_pool2d(x, (2,2)), Tensor.avg_pool2d, gpu=self.gpu, grad_atol=1e-5)
+  def test_maxpool2d(self):
+    for ksz in [(2,2), (3,3), (3,2), (5,5), (5,1)]:
+      for strd in [(1,1), (2,1), (2,2), (4,2)]:
+        with self.subTest(kernel_size=ksz, stride=strd):
```
Comment (geohot) [inline test/test_ops.py:93]: Ooh we should use this in more places

File: tinygrad/opsgpu.py [hunk 2]
```diff
@@ -377,27 +380,31 @@ def backward(ctx, grad_output):
 
 class AvgPool2D(Function):
   @staticmethod
-  def forward(ctx, input, kernel_size=(2, 2)):
+  def forward(ctx, input, kernel_size=(2, 2), stride=None):
```
Comment (geohot) [inline tinygrad/opsgpu.py:383]: stride and kernel_size become ctx.stride and ctx.kernel_size by default

**Comment by geohot** [comment]
Will merge when tests pass.

File: tinygrad/opsgpu.py (hunk 2 above)
Comment (ryanneph) [reply tinygrad/opsgpu.py:383]: Referencing as ctx.kernel_size in the backward pass now.

**Comment by ryanneph** [comment]
Ready to go unless you have any other comments.

**Pull request merged**